packages/edit-site/src/components/block-editor/index.js

```diff
@@ -215,3 +222,41 @@ export default function BlockEditor() {
 		</ExperimentalBlockEditorProvider>
 	);
 }
+
+/**
+ * Returns the appropriate block editor state for a given entity type.
+ *
+ * Note: Navigation entities require a wrapping Navigation block to provide
+ * them with some basic layout and styling. Therefore we create a "ghost" block
+ * and provide it will a reference to the navigation entity ID being edited.
+ *
+ * In this scenario it is the **block** that handles syncing the entity content
+ * whereas for other entities this is handled by entity block editor.
+ *
+ * @param {string} kind the entity kind
+ * @param {string} type the entity type
+ * @return {[WPBlock[], Function, Function]} The block array and setters.
+ */
+function useGetBlockEditorState( kind, type ) {
```

Comment:
This is the key to the PR as it stands.

For entities which don't require a wrapping block (e.g. template parts) we can just entity the entity's blocks directly using the entity block editor.

However, our scenario for Navigation Menus is no conducive to editing the Nav Menu's blocks without a wrapping Nav block. This is principally due to _layout_ requirements.

As a result, instead of passing the Navigation Menu's entities to the BlockEditorProvider instead we create a Nav block with the correct `ref` set and pass _that_ block to the BlockEditorProvider. We then defer to the block to sync any/all edits to the entity.

This feels like a big hack.